packages/babel-helper-fixtures/src/index.ts

```diff
@@ -120,57 +124,97 @@ function pushTask(
+    stdoutLoc,
+    stderrLoc,
+    sourceMap,
+    sourceMapLoc,
+    sourceMapFile,
+    inputSourceMap;
 
-  // If neither input nor exec is present it is not a real testcase
-  if (taskDirStats.isDirectory() && !actualLoc && !execLoc) {
-    if (fs.readdirSync(taskDir).length > 0) {
+  const taskOpts: TaskOptions = JSON.parse(JSON.stringify(suite.options));
+  if (taskDirStats.isDirectory()) {
+    const files = fs.readdirSync(taskDir);
+    for (const file of files) {
+      const name = path.basename(file, path.extname(file));
+      switch (name) {
+        case "input":
+          actualLoc = checkFile(file, false, taskDir, actualLoc);
+          break;
+        case "exec":
+          execLoc = checkFile(file, false, taskDir, execLoc);
+          break;
+        case "output":
+          expectLoc = checkFile(file, true, taskDir, expectLoc);
+          break;
+        case "output.extended":
+          expectLoc = checkFile(file, true, taskDir, expectLoc);
+          break;
+        case "options":
+          taskOptsLoc = `${taskDir}/${file}`;
+          Object.assign(taskOpts, require(taskOptsLoc));
+          break;
+        case "source-map": {
+          sourceMapLoc = `${taskDir}/${file}`;
+          sourceMap = JSON.parse(readFile(sourceMapLoc));
+          sourceMapFile = {
+            loc: sourceMapLoc,
+            code: sourceMap,
+            filename:
+              suiteName + "/" + taskName + "/" + path.basename(sourceMapLoc),
+          };
+          break;
+        }
+        case "input-source-map":
+          inputSourceMap = JSON.parse(readFile(`${taskDir}/${file}`));
+          break;
+      }
+    }
+    // If neither input nor exec is present it is not a real testcase
+    if (files.length > 0 && !actualLoc && !execLoc) {
       console.warn(`Skipped test folder with invalid layout: ${taskDir}`);
+      return;
     }
-    return;
-  } else if (!actualLoc) {
-    actualLoc = taskDir + "/input.js";
-  } else if (!execLoc) {
-    execLoc = taskDir + "/exec.js";
-  }
-
-  const expectLoc =
-    findFile(taskDir + "/output", true /* allowJSON */) ||
-    findFile(`${taskDir}/output.extended`, true) ||
-    taskDir + "/output.js";
-  const stdoutLoc = taskDir + "/stdout.txt";
-  const stderrLoc = taskDir + "/stderr.txt";
-
-  const actualLocAlias =
-    suiteName + "/" + taskName + "/" + path.basename(actualLoc);
-  const expectLocAlias =
-    suiteName + "/" + taskName + "/" + path.basename(actualLoc);
-  let execLocAlias =
-    suiteName + "/" + taskName + "/" + path.basename(actualLoc);
-
-  if (taskDirStats.isFile()) {
+    actualLoc ??= taskDir + "/input.js";
+    execLoc ??= taskDir + "/exec.js";
+    expectLoc ??= taskDir + "/output.js";
+
+    actualLocAlias =
+      suiteName + "/" + taskName + "/" + path.basename(actualLoc);
+    expectLocAlias =
+      suiteName + "/" + taskName + "/" + path.basename(actualLoc);
+    execLocAlias = suiteName + "/" + taskName + "/" + path.basename(actualLoc);
```

Comment:
I know they come from the original code, we can simplify it by the way. :)

Reply:
Good catch. I think they should be different.